pkg/services/provisioning/datasources/provisioner/provisioner.go

```diff
@@ -1,43 +1,64 @@
-package datasources
+package provisioner
 
 import (
+	"context"
 	"errors"
+	"path/filepath"
 
 	"github.com/grafana/grafana/pkg/bus"
-
 	"github.com/grafana/grafana/pkg/infra/log"
-
 	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/registry"
+	"github.com/grafana/grafana/pkg/services/provisioning/datasources"
+	"github.com/grafana/grafana/pkg/services/provisioning/datasources/configreader"
+	"github.com/grafana/grafana/pkg/services/vcs"
+	"github.com/grafana/grafana/pkg/setting"
 )
 
-var (
-	// ErrInvalidConfigToManyDefault indicates that multiple datasource in the provisioning files
-	// contains more than one datasource marked as default.
-	ErrInvalidConfigToManyDefault = errors.New("datasource.yaml config is invalid. Only one datasource per organization can be marked as default")
-)
-
-// Provision scans a directory for provisioning config files
-// and provisions the datasource in those files.
-func Provision(configDirectory string) error {
-	dc := newDatasourceProvisioner(log.New("provisioning.datasources"))
-	return dc.applyChanges(configDirectory)
+// TODO find a way to have the ProvisioningService priority without cyclic import
```

Comment:
I think, currently, what you did is what is done everywhere. But it may be fixed with the work around wire?